javascript/commons/FilterButtons.js

```diff
@@ -211,6 +231,50 @@ liquipedia.filterButtons = {
 				}
 			} );
 		} );
+
+		this.templateExpansions.forEach( ( templateExpansion ) => {
+			const isDefault = templateExpansion.groups.every( ( group ) => {
+				const filterGroup = this.filterGroups[ group ];
+				if ( filterGroup.curated !== filterGroup.defaultCurated ) {
+					return false;
+				}
```

Comment:
That won't work. In your proposal, if `filterGroup.defaultCurated` is false, you'd ignore `filterGroup.curated`

My solution:
curated=false, defaultCurated=false -> proceed to states
curated=true, defaultCurated=false -> return false
curated=false, defaultCurated=true -> return false
curated=true, defaultCurated=true -> proceed to states (could return true here as well?)

Your solution:
curated=false, defaultCurated=false -> proceed to states
curated=true, defaultCurated=false -> proceed to states (!)
curated=false, defaultCurated=true -> return false
curated=true, defaultCurated=true-> return true

Reply:
Honestly i'm also not sure whether we actually want/need this behavior. We could also just call the API, since then we'd also have the potential to get newer data (in cases where the main page is not purged for a while)

Reply:
I'll leave that to @FO-nTTaX and @liquidely

Reply:
I'd rather not call the API again if it's a result we've already had.

At the moment we call the API every time some button is clicked unless it's default, but we should not even call the API again if we've already had a certain combination before. So I would argue we should do even better with caching instead of having less of it.

At the moment on the test page, if I only have S-Tier enabled, then enable A_Tier and then disable A-Tier again I do the same template call for only S-Tier again, even though we should have had that data already. One somewhat lazy but effective solution would be to cache based on a hash of the object that is the first parameter to `api.get()` and just reuse the result of that if we had already grabbed that data before.

Reply:
Particularly for logged in users it won't necessarily, it also depends on browser settings and if you're in a private window or not. Adding our own caching layer into a local object or whatever circumvents all these differences.

Reply:
In particular some privacy addons will disable the cache for xhr resources.

Reply:
How long would you want to cache the results in local storage?

Reply:
I would not persist this cache between page refreshes, just as long as the page is open should be fine imo. I would not overthink this caching as most people won't be constantly F5ing the pages with these filters or changing the filters all the time.